### PR TITLE
Apache Ignite based cluster manager implementation for Vert.x

### PR DESCRIPTION
**Pull request by agura**
Signed-off-by: Andrey Gura <andreyn.gura@gmail.com>

### CODE REVIEW

File: src/main/asciidoc/java/index.adoc [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+Add a dependency to the artifact.
+
+* Maven (in your `pom.xml`):
+
+[source,xml,subs="+attributes"]
+----
+<dependency>
+  <groupId>io.vertx</groupId>
+  <artifactId>vertx-ignite</artifactId>
+  <version>3.2.0</version>
```
Comment (cescoffier) [inline src/main/asciidoc/java/index.adoc]: It should be regenerated, 3.3.0-SNAPSHOT is expected

File: src/main/asciidoc/java/index.adoc [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+----
+org.apache.ignite.level=INFO
+----
+
+and also
+
+----
+java.util.logging.ConsoleHandler.level=INFO
+java.util.logging.FileHandler.level=INFO
+----
```
Comment (cescoffier) [inline src/main/asciidoc/java/index.adoc]: As it seems to be a recurrent topic, could we add a section about encryption, if it's supported by Ignite ?

File: src/main/java/io/vertx/spi/cluster/ignite/IgniteClusterManager.java [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.apache.ignite.events.EventType.*;
+
+/**
+ * Apache Ignite based cluster manager.
+ */
```
Comment (cescoffier) [inline src/main/java/io/vertx/spi/cluster/ignite/IgniteClusterManager.java]: @author required

File: src/main/java/io/vertx/spi/cluster/ignite/impl/AsyncMapImpl.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.shareddata.AsyncMap;
+import java.util.function.Consumer;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.lang.IgniteFuture;
+
+/**
+ * Async wrapper for {@link MapImpl}.
+ */
```
Comment (cescoffier) [inline src/main/java/io/vertx/spi/cluster/ignite/impl/AsyncMapImpl.java]: @author required

File: src/main/java/io/vertx/spi/cluster/ignite/impl/AsyncMultiMapImpl.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+/**
+ * MultiMap implementation.
+ */
```
Comment (cescoffier) [inline src/main/java/io/vertx/spi/cluster/ignite/impl/AsyncMultiMapImpl.java]: @Author required

File: src/main/java/io/vertx/spi/cluster/ignite/impl/ChoosableIterableImpl.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+import io.vertx.core.spi.cluster.ChoosableIterable;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * ChoosableIterable implementation.
+ */
```
Comment (cescoffier) [inline src/main/java/io/vertx/spi/cluster/ignite/impl/ChoosableIterableImpl.java]: @author required

File: src/main/java/io/vertx/spi/cluster/ignite/impl/MapImpl.java [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Represents Apache Ignite cache as {@link java.util.Map} interface implementation.
+ */
```
Comment (cescoffier) [inline src/main/java/io/vertx/spi/cluster/ignite/impl/MapImpl.java]: @Author required

**Comment by cescoffier** [comment]
A couple of comment. Once fixed in can be merged in `master`.

File: src/main/asciidoc/java/index.adoc (hunk 2 above)
Comment (agura) [inline src/main/asciidoc/java/index.adoc]: Yes, Ignite supports encryption. May be better place is default configuartion file with commented out piece of configuration with documentaton link?

**Comment by agura** [comment]
@cescoffier 

Clement, I fixed all your comments. Default Ignite configuration was updated in order to provide example of SSL\TLS encryption configuration.

File: src/main/asciidoc/java/index.adoc (hunk 2 above)
Comment (cescoffier) [inline src/main/asciidoc/java/index.adoc]: Yes, we do that for hazelcast. That would be great to have this.

**Comment by cescoffier** [comment]
:+1: 

@vietj @pmlopes fancy a review ?

**Comment by pmlopes** [comment]
:+1:

**Pull request merged**